docs/testplan/subnet_decap/Subnet-decap-test-plan.md

```diff
@@ -0,0 +1,181 @@
+![Subnet decap](./vlan_subnet_probing.png)
+
+
+## Topology
+
+Supported topologies: T0 and T1
+
+## Testbed setup
+
+The test requires subnet decap being enabled on the DUT. It can be enabled by injecting below configuration into CONFIG_DB.
+```
+{
+    "SUBNET_DECAP|subnet_type": {
+        "status": "enable",
+        "src_ip": "20.20.20.0/24",
+        "src_ip_v6": "fc01::0/120"
+    }
+}
+```
+On T0 topology, the default vlan (Vlan1000) is required for testing. The IP of Vlan1000 is `192.168.0.1/21` and `fc02:1000::1/64`.
+## Test cases
+
+### Test case 1 - Verify IPinIP packet targeting at VLAN subnet is decapsulated (T0 only)
+#### Test steps
+#### IPV4 positive scenario
+1. Craft `IPinIP` packet with signature as below.
+
+  - Outer DIP: Any IP in `192.168.0.1/21`
+  - Outer SIP: Any IP in `20.20.20.0/24`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+
+2. Ingress the packet from upstream ports to DUT.
+3. Verify packet is decapsulated and routed to inner DIP.
+
+#### IPV4 negative scenario
+1. Craft `IPinIP` packet with signature as below.
+
+  - Outer DIP: Any IP in `192.168.0.1/21`
+  - Outer SIP: Any IP **NOT** in `20.20.20.0/24`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+2. Setup ARP responder on ptf and trigger MAC learning with `ping` from DUT.
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is forwarded to expected VLAN port.
+
+#### IPV6 positive scenario
+1. Craft `IPV6inIPV6` packet with signature as below.
+
+  - Outer DIP: Any IP in `fc02:1000::1/64`
+  - Outer SIP: Any IP in `fc01::0/120`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+
+2. Ingress the packet from upstream ports to DUT.
+3. Verify packet is decapsulated and routed to inner DIP.
+
+#### IPV6 negative scenario
+1. Craft `IPV6inIPV6` packet with signature as below.
+
+  - Outer DIP: Any IP in `fc02:1000::1/64`
+  - Outer SIP: Any IP **NOT** in `fc01::0/120`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+
+2. Setup ARP responder on ptf and trigger NBR discovery with `ping` from DUT.
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is forwarded to expected VLAN port.
+
+### Test case 2 - Verify IPinIP packet targeting at VIP is decapsulated (T0 only)
+#### Test setup
+1. Setup SLB BGP neighbor on ptf and establish BGP sessions with DUT.
+2. Announce some routes (including both IPv4 and IPv6) to DUT from SLB.
+#### Test steps
+#### IPV4 scenario
+1. Verify `STATE_DB` entries are created as expected.
+2. Craft packet with below signature
```

Comment:
better to validate by sending IPinIP packets with source IP not in the range `20.20.20.0/24`

Reply:
Why? For VIP decap we are also checking the src_ip.